### PR TITLE
Remove direct dependency on "zend-diactoros"

### PR DESCRIPTION
**Pull request by spideyfusion**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,19 +16,19 @@
         "doctrine/doctrine-bundle": "^1.8",
         "doctrine/orm": "^2.6",
         "league/oauth2-server": "^7.2",
-        "sensio/framework-extra-bundle": "^3.0.0|^4.0.0|^5.0.0",
-        "symfony/framework-bundle": "~3.4|~4.0",
-        "symfony/psr-http-message-bridge": "^1.0",
-        "symfony/security-bundle": "~3.4|~4.0",
-        "zendframework/zend-diactoros": "^1.7|^2.1"
+        "nyholm/psr7": "^1.1",
```
Comment (ajgarlag) [inline composer.json]: I think `nyholm/psr7` should be moved to `require-dev` and `psr/http-factory` should explicitly required here

File: Tests/Integration/AbstractIntegrationTest.php [hunk 2]
```diff
@@ -150,7 +150,7 @@ protected function createAuthorizationRequest(?string $credentials, array $body
             'Authorization' => sprintf('Basic %s', base64_encode($credentials)),
         ];
 
-        return new ServerRequest([], [], null, null, 'php://temp', $headers, [], [], $body);
+        return (new ServerRequest('', '', $headers))->withParsedBody($body);
```
Comment (ajgarlag) [inline Tests/Integration/AbstractIntegrationTest.php]: Why not use `Nyholm\Psr7\Factory\Psr17Factory` to build PSR-7 instances required in tests?

File: README.md [hunk 3]
```diff
@@ -26,17 +26,19 @@ This package is currently in the active development.
...
 
     ```sh
-    composer require trikoder/oauth2-bundle --no-plugins --no-scripts
+    composer require trikoder/oauth2-bundle nyholm/psr7 --no-plugins --no-scripts
     ```
 
-    > **NOTE:** Due to required pre-configuration, this bundle is currently not compatible with [Symfony Flex](https://github.com/symfony/flex).
+    > **NOTE #1:** Due to required pre-configuration, this bundle is currently not compatible with [Symfony Flex](https://github.com/symfony/flex).
+
+    > **NOTE #2:** This bundle requires a PSR 7/17 implementation to operate. We recommend that you use [nyholm/psr7](https://github.com/Nyholm/psr7). Check out this [document](docs/psr-implementation-switching.md) if you with to use a different implementation.
```
Comment (X-Coder264) [inline README.md]: @spideyfusion "With" -> "wish" typo 😛 The rest of the PR looks good (as we had discussed) 👍

File: composer.json (hunk 1 above)
Comment (spideyfusion) [reply composer.json]: @ajgarlag Yeah, it didn't make sense to force the user to use a specific PSR-7/17 implementation. The `psr/http-factory` is now explicitly required and I've added instructions on how to switch between different implementations.

File: Tests/Integration/AbstractIntegrationTest.php (hunk 2 above)
Comment (spideyfusion) [reply Tests/Integration/AbstractIntegrationTest.php]: @ajgarlag Done.

File: .travis.yml [hunk 4]
```diff
@@ -29,7 +31,7 @@ before_script:
   - dev/bin/php composer install --ansi --prefer-dist
 
 script:
-  - dev/bin/php composer test -- --colors=always --coverage-clover=coverage.xml --debug
+  - PSR_HTTP_PROVIDER=${PSR_HTTP_PROVIDER} dev/bin/php composer test -- --colors=always --coverage-clover=coverage.xml --debug
```
[the remark this answers is not in the document]
Comment (spideyfusion) [reply .travis.yml]: @HypeMC Good catch.

File: README.md (hunk 3 above)
Comment (spideyfusion) [reply README.md]: @X-Coder264 Good catch.

**Pull request merged**